server/galaxyls/tests/files/complex_inputs_01.xml

```diff
@@ -0,0 +1,59 @@
+<tool id="complex_inputs_01" name="Complex inputs">
+    <inputs>
+        <conditional name="c1">
+            <param name="c1_action" type="select">
+                <option value="a1" selected="True">A 1</option>
+                <option value="a2">A 2</option>
+                <option value="a3">A 3</option>
+            </param>
+            <when value="a1">
+                <param name="c1_a1_p1" type="text"/>
+                <section name="sec1" title="C1 A1 Section" expanded="true">
+                    <param name="number" value="1" type="integer"/>
+                </section>
+            </when>
+            <when value="a2">
+                <param name="c1_a2_p1" type="text"/>
+                <conditional name="c2">
+                    <param name="c2_action" type="select">
+                        <option value="c2a1" selected="True">A 1</option>
+                        <option value="c2a2">A 2</option>
+                    </param>
+                    <when value="c2a1">
+                        <param name="c2_a1_p1" type="text"/>
+                        <section name="sec_c2a1" title="C2 A1 Section" expanded="true">
+                            <section name="sub_sec_c2a1" title="C2 A1 Section" expanded="true">
+                                <param name="sub_p01" value="3.2" type="float"/>
+                                <param name="sub_p02" value="4" type="integer"/>
+                                <param name="sub_p03" type="boolean" truevalue="--sub_p03" falsevalue="" checked="false" />
```

Comment:
I changed that since booleans have truevalue and falsevalue. Wondering why your linter has not caught that. Maybe its missing in the CTD.

Reply:
The reason the linter can not catch this is that, in the [XSD](https://github.com/galaxyproject/galaxy/blob/c3083053318c09274e5b0397d9285121eaf22c4d/lib/galaxy/tool_util/xsd/galaxy.xsd#L2656), the only attribute that is really required in a `param` element is the `type`. I understand that this kind of restriction to have different attributes required depending on one of the other attributes (in this case `type`) is not easy to (or can not be) modeled with the schema.
I believe, if necessary, these kinds of checks can also be done directly in the language-server.